lib/spack/llnl/util/cpu/detect.py

```diff
@@ -126,8 +127,8 @@ def raw_info_dictionary():
     for factory in info_factory[platform.system()]:
         try:
             info = factory()
-        except Exception:
-            pass
+        except Exception as e:
+            warnings.warn(e)
```

Comment:
@alalazo This works as expected:
```console
$ spack arch
==> Warning: can only concatenate str (not "bytes") to str
darwin-mojave-x86_64
```
However, I'm not sure how to make it a "debug-level" message.

Reply:
Is there any way we can make the Exception catching less generic?

Reply:
Well, I am already catching `Exception` and not `BaseException`...